gloo/docs/user_guides/upstreams/aws_upstream.md

```diff
@@ -0,0 +1,130 @@
+---
+title: AWS Upstream
+weight: 1
+description: AWS Upstream configuration guide.
+---
+
+## How to setup and use AWS Upstream
+
+There are 2 steps to enabling Gloo to discover and access AWS Lambda services.
+
+1. Create an AWS Secret to give Gloo credentials to access AWS.
+2. Create a Gloo upstream, referencing AWS Secret, that will populate the Gloo function catalog with available
+AWS Lambda functions. 
+
+### Create AWS Secret
+
+The following command will create a Kubernetes secret that contains the AWS Access Key and Secret Key needed by Gloo
+to connect to AWS for service discovery.
+
+```shell
+glooctl create secret aws --help
+
+Create an AWS secret with the given name
+
+Usage:
+  glooctl create secret aws [flags]
+
+Flags:
+      --access-key string   aws access key
+  -h, --help                help for aws
+      --name string         name of the resource to read or write
+  -n, --namespace string    namespace for reading or writing resources (default "gloo-system")
+      --secret-key string   aws secret key
+
+Global Flags:
+  -i, --interactive     use interactive mode
+  -o, --output string   output format: (yaml, json, table)
+```
+
+For example, to create an AWS secret named `my-aws` in the (default) namespace `gloo-system`, run the following command.
+You can name the secret (`--name 'your_name'`) whatever you like. Just make sure you use the correct name when
+referencing it from AWS Upstream.
+
+```shell 
+glooctl create secret aws \
+    --name 'my-aws' \
+    --namespace gloo-system \
+    --access-key '<AWS ACCESS KEY>' \
+    --secret-key '<AWS SECRET KEY>'
+```
+
+You can see the details of the created secret as follows.
+
+```shell
+kubectl describe secret my-aws
+```
+
+```noop
+Name:         my-aws
+Namespace:    gloo-system
+Labels:       <none>
+Annotations:  resource_kind: *v1.Secret
+
+Type:  Opaque
+
+Data
+====
+aws:  84 bytes
+```
+
+### Create AWS Upstream
+
+Then to create an AWS Upstream so that Gloo can both do Lambda service discovery, and to allow you to create
```

Comment:
nit: sentence is a bit hard to read; service -> function/endpoint

This command can be used to create an AWS upstream. Once it is created, Gloo can do Lambda function discovery, and you can create virtual services with route rules referencing Lambda functions.